### PR TITLE
Unskip tests for driver adapters core features, skip tests for low fidelity cases

### PR DESCRIPTION
**Pull request by miguelff**
Closes https://github.com/prisma/team-orm/issues/472
Closes https://github.com/prisma/team-orm/issues/520

Sibling to https://github.com/prisma/prisma/pull/21889

### CODE REVIEW

File: .github/workflows/query-engine-driver-adapters.yml [hunk 1]
```diff
@@ -31,8 +31,6 @@ jobs:
             setup_task: 'dev-neon-ws-postgres13'
           - name: 'libsql'
             setup_task: 'dev-libsql-sqlite'
-          - name: 'planetscale'
-            setup_task: 'dev-planetscale-vitess8'
```
Comment (janpio) [inline .github/workflows/query-engine-driver-adapters.yml:35]: Intentionally dropping something that is currently in `main`?

File: query-engine/connector-test-kit-rs/query-engine-tests/tests/new/regressions/prisma_15204.rs [hunk 2]
```diff
@@ -38,8 +38,22 @@ mod conversion_error {
...
-    async fn convert_to_bigint(runner: Runner) -> TestResult<()> {
+    #[connector_test(schema(schema_int), only(Sqlite, JS))]
+    async fn convert_to_int_sqlite_js(runner: Runner) -> TestResult<()> {
+        create_test_data(&runner).await?;
+
+        assert_error!(
+            runner,
+            r#"query { findManyTestModel { field } }"#,
+            2023,
+            "Inconsistent column data: Conversion failed: number must be an integer in column 'field'"
```
Comment (janpio) [inline query-engine/connector-test-kit-rs/query-engine-tests/tests/new/regressions/prisma_15204.rs:49]: We can not easily make this error message similar to the Quaint one?

(It's fine, just and error edge case, but double checking as of course less variants would be better)

File: query-engine/connector-test-kit-rs/query-engine-tests/tests/raw/sql/casts.rs [hunk 3]
```diff
@@ -5,7 +5,20 @@ use query_engine_tests::*;
 mod casts {
     use query_engine_tests::{fmt_query_raw, run_query, RawParam};
 
-    #[connector_test]
+    // The following tests are excluded for driver adapters. The underlying
+    // driver rejects queries where the values of the positional arguments do
+    // not match the expected types. As an example, the following query to the
+    // driver
```
Comment (janpio) [inline query-engine/connector-test-kit-rs/query-engine-tests/tests/raw/sql/casts.rs:11]: Does that mean that our raw query API with driver adapters is less accepting than our Rust one?

(Not a problem at all, just interesting)

**Review by janpio — APPROVED** [review]

File: query-engine/connector-test-kit-rs/query-engine-tests/tests/raw/sql/casts.rs (hunk 3 above)
Comment (miguelff) [reply query-engine/connector-test-kit-rs/query-engine-tests/tests/raw/sql/casts.rs:11]: Yes, it's a little bit more strict.

File: .github/workflows/query-engine-driver-adapters.yml (hunk 1 above)
Comment (miguelff) [reply .github/workflows/query-engine-driver-adapters.yml:35]: Yes, this doesn't provide any value at all until fituring out the vitess test situation for driver adapters. Which has it's own issue. As of now, the feedback on this test suite doesn't provide any kind of value.

BTW: PS tests pass on prisma/prisma, but I expect a lot of tests broken for planetscale in the engines.

File: query-engine/connector-test-kit-rs/query-engine-tests/tests/new/regressions/prisma_15204.rs (hunk 2 above)
Comment (miguelff) [reply query-engine/connector-test-kit-rs/query-engine-tests/tests/new/regressions/prisma_15204.rs:49]: Not exactly the same. No.

**Comment by miguelff** [comment]
This merge commit will fail, because it requires changes from https://github.com/prisma/prisma/pull/21889, it's OK.
